### PR TITLE
use retryable txns throughout token bridge

### PR DESCRIPTION
**Pull request by DZGoldman**

### CODE REVIEW

File: packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol [hunk 1]
```diff
@@ -101,16 +102,27 @@ contract EthERC20Bridge is L1Buddy {
...
         address l2Address = customL2Tokens[l1Address];
         require(l2Address != address(0), "NOT_REGISTERED");
-        sendPairedContractTransaction(
-            maxGas,
-            gasPriceBid,
-            abi.encodeWithSignature("customTokenRegistered(address,address)", l1Address, l2Address)
-        );
+        return
+            inbox.createRetryableTicket{ value: msg.value }(
+                l2Address,
```
Comment (DZGoldman) [inline packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol:113]: this should be l2Buddy

File: packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol [hunk 2]
```diff
@@ -101,16 +102,27 @@ contract EthERC20Bridge is L1Buddy {
...
-    ) external payable {
+    ) external payable returns (uint256) {
         address l2Address = customL2Tokens[l1Address];
         require(l2Address != address(0), "NOT_REGISTERED");
-        sendPairedContractTransaction(
-            maxGas,
-            gasPriceBid,
-            abi.encodeWithSignature("customTokenRegistered(address,address)", l1Address, l2Address)
-        );
+        return
```
Comment (hkalodner) [inline packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol:111]: I think we want an event emitted here including the msg num from the `createRetryableTicket` call. `ActivatedCustomToken` or something like that

File: packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol [hunk 3]
```diff
@@ -176,11 +189,18 @@ contract EthERC20Bridge is L1Buddy {
                 ? ArbTokenBridge.updateERC777TokenInfo.selector
                 : ArbTokenBridge.updateERC20TokenInfo.selector;
 
-        sendPairedContractTransaction(
-            maxGas,
-            gasPriceBid,
-            abi.encodeWithSelector(_selector, erc20, name, symbol, decimals)
-        );
+        bytes memory data = abi.encodeWithSelector(_selector, erc20, name, symbol, decimals);
+        return
```
Comment (hkalodner) [inline packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol:193]: Similar to above. `UpdatedTokenInfo` including the inbox msg num

File: packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol [hunk 4]
```diff
@@ -203,7 +223,7 @@ contract EthERC20Bridge is L1Buddy {
 
         bytes memory data = abi.encodeWithSelector(selector, erc20, destination, amount, decimals);
         return
-            inbox.createRetryableTicket(
```
Comment (hkalodner) [inline packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol:206]: Add event

File: packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol [hunk 5]
```diff
@@ -277,7 +297,7 @@ contract EthERC20Bridge is L1Buddy {
                 amount
             );
         uint256 seqNum =
-            inbox.createRetryableTicket(
+            inbox.createRetryableTicket{ value: msg.value }(
```
Comment (hkalodner) [inline packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol:300]: Add event

File: packages/arb-bridge-peripherals/contracts/buddybridge/ethereum/L1Buddy.sol [hunk 6]
```diff
@@ -58,36 +58,37 @@ abstract contract L1Buddy {
...
         codeHash = keccak256(contractInitCode);
-        l2Buddy = BuddyUtil.calculateL2Address(
-            address(l2Deployer),
-            address(this),
-            codeHash
-        );
+        l2Buddy = BuddyUtil.calculateL2Address(address(l2Deployer), address(this), codeHash);
         l2Connection = L2Connection.Initiated;
+
+        return
```
Comment (hkalodner) [inline packages/arb-bridge-peripherals/contracts/buddybridge/ethereum/L1Buddy.sol:78]: Add `BuddyContractDeployed` or something like that

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol (hunk 4 above)
Comment (DZGoldman) [reply packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/EthERC20Bridge.sol:206]: this one one already has DepositERC20 and DepositERC777 where it gets called

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**